dashboard/test/models/concerns/pd/form_test.rb

```diff
@@ -2,9 +2,39 @@
 
 class DummyForm
   include ActiveModel::Validations
+  include ActiveModel::Dirty
   include Pd::Form
 
+  # minimal implementation from
+  # http://api.rubyonrails.org/classes/ActiveModel/Validations.html
   attr_accessor :form_data
+
+  # minimal implementation from
+  # http://api.rubyonrails.org/classes/ActiveModel/Dirty.html
+  define_attribute_methods :form_data
+
+  def initialize
+    @form_data = nil
+  end
+
+  def form_data=(val)
+    form_data_will_change! unless val == @form_data
+    @form_data = val
+  end
+
+  def save
+    # do persistence work
+    changes_applied
+  end
+
+  def reload!
+    # get the values from the persistence layer
+    clear_changes_information
+  end
+
+  def rollback!
+    restore_attributes
+  end
```

Comment:
The problem is that I'd have to actually create a table for the dummy class, otherwise I get

```
Pd::FormTest
  test_pd_form_enforces_required_fields                          ERROR (0.01s)
ActiveRecord::StatementInvalid:         ActiveRecord::StatementInvalid: Mysql2::Error: Table 'dashboard_test.dummy_forms' doesn't exist: SHOW FULL FIELDS FROM `dummy_forms`
            test/models/concerns/pd/form_test.rb:43:in `block in <class:FormTest>'
            test/testing/setup_all_and_teardown_all.rb:22:in `run'
```

Let me poke around and see if there's a way I can inherit from `ActiveRecord::Base` and just tell it not to try to do that; I agree it would be much cleaner.